packages/http-client-csharp/emitter/src/lib/operation-converter.ts

```diff
@@ -318,22 +321,86 @@ function getMediaTypes(type: SdkType): string[] {
 }
 
 function loadOperationPaging(
+  context: CSharpEmitterContext,
   method: SdkServiceMethod<SdkHttpOperation>,
+  rootApiVersions: string[],
+  uri: string,
 ): OperationPaging | undefined {
-  if (method.kind !== "paging" || method.__raw_paged_metadata === undefined) {
+  if (method.kind !== "paging" || method.pagingMetadata === undefined) {
     return undefined;
   }
 
+  let nextLink: NextLink | undefined;
+  if (method.pagingMetadata.nextLinkSegments) {
+    nextLink = {
+      ResponseSegments: method.pagingMetadata.nextLinkSegments.map((segment) =>
+        getResponseSegmentName(segment),
+      ),
+      ResponseLocation: getResponseLocation(method.pagingMetadata.nextLinkSegments[0]),
+    };
+
+    if (method.pagingMetadata.nextLinkOperation) {
+      nextLink.Operation = fromSdkServiceMethod(
+        context,
+        method.pagingMetadata.nextLinkOperation,
+        uri,
+        rootApiVersions,
+      );
+    }
+  }
+
+  let continuationToken: ContinuationToken | undefined;
+
+  if (
+    method.pagingMetadata.continuationTokenParameterSegments &&
+    method.pagingMetadata.continuationTokenResponseSegments
+  ) {
+    // The last segment of the service method parameter can be used to map back to the protocol parameter
+    const lastParameterSegment = method.pagingMetadata.continuationTokenParameterSegments[
+      method.pagingMetadata.continuationTokenParameterSegments.length - 1
+    ] as SdkModelPropertyType;
+    continuationToken = {
+      Parameter: fromSdkHttpOperationParameter(
+        context,
+        getHttpOperationParameter(method, lastParameterSegment)!,
+        rootApiVersions,
+      ),
+      ResponseSegments: method.pagingMetadata.continuationTokenResponseSegments!.map((segment) =>
+        getResponseSegmentName(segment),
+      ),
+      ResponseLocation: getResponseLocation(
+        method.pagingMetadata.continuationTokenResponseSegments?.[0],
+      ),
+    };
+  }
+
   return {
-    ItemName: method.__raw_paged_metadata.itemsProperty?.name,
-    NextLinkName: method.__raw_paged_metadata.nextLinkProperty?.name,
+    // TODO - this is hopefully temporary until TCGC provides the information directly on pagingMetadata https://github.com/Azure/typespec-azure/issues/2291
+    ItemPropertySegments: method.response.resultSegments!.map((s) => s.name),
+    NextLink: nextLink,
+    ContinuationToken: continuationToken,
```

Comment:
so there was a continuous effort on aligning our models with the models defined in TCGC so that in the future, when we get everything aligned, our emitter could do nothing but simply serializes everything we get from TCGC into the tspCodeModel.json file.
Therefore could we align our paging information with theirs in your PR if possible?

Reply:
The paging data from TCGC doesn't have all the info that we will need to generate. Also, there are open issues about changing the TCGC shape so that may be changing anyway.

Reply:
Discussed offline and our InputModelProperty doesn't really align with SdkModelPropertyType since SdkModelPropertyType also includes headers, parameters, etc. 

For now, we will go with the minimum data required to generate, and the work to align with TCGC will be ongoing.